cypress/CYPRESS.md

```diff
@@ -8,16 +8,106 @@ SEE: [README](../README.md) for instructions on setting up and running tests
 
 To get total coverage, run both e2e and component tests.
 
-```
+```bash
 npm run cypress:run
 ```
 
 Open the coverage report
 
-```
+```bash
 npm run coverage
 ```
 
+## End-to-End Tests Guidelines
+
+End-to-End tests for our project are located in the `cypress/e2e` directory. The following guidelines are designed to ensure that our E2E tests are robust, maintainable, and effective.
+
+### General Principles
+
+1. **Test Isolation and Independence**:
+
+   - Each test should be independent and capable of running on its own without relying on the state created by previous tests.
+   - If during deployment certain resources were created, those should not be deleted as part of the test runs.
+   - Avoid tests that require execution in a specific order.
+
+2. **Environment**:
+
+   - Tests should be executable in any environment without depending on the specific state of that environment.
+   - Design tests to be environment-agnostic, ensuring they can run in development and production environments.
+
+3. **Resource Management**:
+   - Each test should create all necessary resources for its execution and clean them up after completion.
+   - This includes data creation, user logins, and any other setup or teardown tasks.
+   - Leveraging the `before` and `after` hooks in Cypress can help manage resource creation and cleanup.
+
+### Test Implementation
+
+1. **Element Selection**:
+
+   - Prefer using data attributes (like `data-cy`) for selecting elements.
+   - Avoid using CSS classes or IDs for element selection, as they can change frequently and are not unique.
+
+2. **Asynchronous Operations**:
+
+   - Properly handle the asynchronous nature of web applications, using Cypress's automatic waiting features or custom waits when necessary.
+   - Avoid to use `cy.wait` as much as possible - mainly with hardcoded wait values, prefer to use `cy.get` with assertions.
+
+   **Example**:
+
+   ```javascript
+   describe('User Profile Test', () => {
+     it('loads user profiles asynchronously', () => {
+       // Visit the user profile page
+       cy.visit('/profiles');
+
+       // Interact with an element that triggers an asynchronous operation
+       cy.get('button.load-profiles').click();
+
+       // Cypress automatically waits for the element to appear in the DOM
+       cy.get('.profile-list').should('be.visible');
+
+       // Assert on the elements loaded as a result of the asynchronous operation
+       cy.get('.profile-list .profile-item').should('have.length', 5);
+
+       // You can also wait for specific text to appear
+       cy.contains('.profile-list', 'John Doe');
+
+       // Asserting on elements that are expected to update
+       cy.get('.profile-status').should('contain', 'Profiles Loaded');
+     });
+   });
+   ```
+
+3. **Test Data**:
+   - Utilize dynamic data creation methods or fixtures to avoid dependency on hard-coded data.
```

Comment:
fixtures should ideally only be used in component tests. E2E tests should communicate with the live API.